substrate/frame/staking-async/ahm-test/src/ah/mock.rs

```diff
@@ -571,3 +660,119 @@ pub(crate) fn election_events_since_last_call() -> Vec<multi_block::Event<T>> {
 	ElectionEventsIndex::set(all.len());
 	all.into_iter().skip(seen).collect()
 }
+
+pub(crate) fn roll_and_assert_idle_sessions(count: usize, expect_export: bool) {
+	let last_session_end_index =
+		pallet_staking_async_rc_client::LastSessionReportEndingIndex::<T>::get()
+			.unwrap_or_default();
+	let active_era = ActiveEra::<T>::get().unwrap().index;
+	let planning_era = CurrentEra::<T>::get().unwrap();
+	let pre_validator_set_export_count = LocalQueue::get().unwrap().len();
+
+	for s in 1..=count {
+		// end the session..
+		assert_ok!(pallet_staking_async_rc_client::Pallet::<T>::relay_session_report(
+			RuntimeOrigin::root(),
+			SessionReport {
+				end_index: last_session_end_index + s as u32,
+				validator_points: vec![(1, 10)],
+				activation_timestamp: None,
+				leftover: false,
+			}
+		));
+
+		// run session blocks.
+		roll_many(60);
+
+		// should be no change in active or planning era
+		assert_eq!(ActiveEra::<T>::get().unwrap().index, active_era);
+		assert_eq!(CurrentEra::<T>::get().unwrap(), planning_era);
+		if expect_export {
+			// new validator set exported
+			assert_eq!(LocalQueue::get().unwrap().len(), pre_validator_set_export_count + 1);
+		} else {
+			// no new messages in the queue
+			assert_eq!(LocalQueue::get().unwrap().len(), pre_validator_set_export_count);
+		}
+	}
+}
+
+pub(crate) fn roll_and_assert_election_session(expect_export: bool) {
+	// clear election events
+	election_events_since_last_call();
+
+	let last_session_end_index =
+		pallet_staking_async_rc_client::LastSessionReportEndingIndex::<T>::get()
+			.unwrap_or_default();
+	let active_era = ActiveEra::<T>::get().unwrap().index;
+	let planning_era = CurrentEra::<T>::get().unwrap();
+	let pre_validator_set_export_count = LocalQueue::get().unwrap().len();
+	// election should not have already started.
+	assert_eq!(active_era, planning_era);
+	assert_eq!(multi_block::CurrentPhase::<T>::get(), Phase::Off);
+
+	assert_ok!(pallet_staking_async_rc_client::Pallet::<T>::relay_session_report(
+		RuntimeOrigin::root(),
+		SessionReport {
+			end_index: last_session_end_index + 1,
+			validator_points: vec![(1, 10)],
+			activation_timestamp: None,
+			leftover: false,
+		}
+	));
+
+	// assert no change in active era
+	assert_eq!(ActiveEra::<T>::get().unwrap().index, active_era);
+	// planning era is incremented indicating start of election
+	assert_eq!(CurrentEra::<T>::get().unwrap(), planning_era + 1);
+	assert_eq!(
+		election_events_since_last_call(),
+		[ElectionEvent::PhaseTransitioned { from: Phase::Off, to: Phase::Snapshot(Pages::get()) }]
+	);
+
+	roll_many(60);
```

Comment:
There is a function in `ElectionProvider` called `election_duration` that you might be able to use as a replacement here. 

Just a bit cautious of the magic `roll_many(60)` in the code and how they can be improved. 

Also, you can do `loop` and `roll_next()` in it until `Phase::Off` is met again, with some upper bound?

Reply:
done